packages/router/src/params.tsx

```diff
@@ -26,19 +25,21 @@ export const ParamsProvider: React.FC = ({ children }) => {
         paramTypes
       )
 
-      if (match) {
-        initialParams = {
-          ...initialParams,
-          ...params,
-        }
+      if (match && typeof params !== 'undefined') {
+        pathParams = params
       }
     }
   }
 
-  const [params, setParams] = useState<Record<string, string>>(initialParams)
-
   return (
-    <ParamsContext.Provider value={{ params, setParams }}>
+    <ParamsContext.Provider
+      value={{
+        params: {
+          ...searchParams,
+          ...pathParams,
```

Comment:
I like that we've changed the order here. This is related to https://github.com/redwoodjs/redwood/issues/2121

Should probably have something about this in the release notes

Reply:
Good idea, I'll start writing those up!

Reply:
I've changed this back to the original order, going to keep this the same in this release.

Reply:
Let's revisit for 0.29. We can continue the discussion in #2121, or in a separate PR connected to that issue